### PR TITLE
remove null checks for key/value in write_object_value

### LINKED ISSUE #14: FormSerializationWriter does not actually write content to the request
**Environment**

- Python Version
3.11.2
- kiota-serialization-form version:
0.1.0
- OS:
Debian 12.5 (via docker)

**Stack trace (if available)**
N/A

**Describe the bug**
(I'm not sure if this is a bug, or just "we haven't gotten to this yet")
When using the FormSerializationWriter, it does not actually write any content to the RequestInformation.

**To Reproduce**
Generate a python client with the provided yaml and run the provided script.  It will demonstrate that the body is not written to the request.
[bug-openapi.txt](https://github.com/microsoft/kiota-serialization-form-python/files/14589473/bug-openapi.txt)
[fsw-bug.txt](https://github.com/microsoft/kiota-serialization-form-python/files/14589474/fsw-bug.txt)

**Expected behavior**
The body object should be in the content

**Screenshots**

**Additional context**
I believe the problem is because at
https://github.com/microsoft/kiota-abstractions-python/blob/main/kiota_abstractions/request_information.py#L179
the 'write_object_value' method is called with a None key and the body object, but in the FormSerializationWriter
https://github.com/microsoft/kiota-serialization-form-python/blob/main/kiota_serialization_form/form_serialization_writer.py#L189
a None key means that nothing at all happens.  The documentation indicates the key is supposed to be optional.

### PR DESCRIPTION
**Pull request by shemogumbe**
Fixes https://github.com/microsoft/kiota-serialization-form-python/issues/14

## Overview

Brief description of what this PR does, and why it is needed.

## Related Issue

Fixes # (issue)

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

* How to test this PR
* Prefer bulleted description
* Start after checking out this branch
* Include any setup required, such as bundling scripts, restarting services, etc.
* Include test case, and expected output

### CODE REVIEW

File: kiota_serialization_form/form_serialization_writer.py [hunk 1]
```diff
@@ -186,24 +186,23 @@ def write_object_value(
             additional_values_to_merge (tuple[Parsable]): The additional values to merge to the
             main value when serializing an intersection wrapper.
         """
-        if key and (value or additional_values_to_merge):
-            temp_writer = self._create_new_writer()
+        temp_writer = self._create_new_writer()
 
-            if value:
-                self._serialize_value(temp_writer, value)
+        if  value is not None:
```
Comment (baywet) [inline kiota_serialization_form/form_serialization_writer.py]:
```suggestion
        if value is not None:
```

File: kiota_serialization_form/form_serialization_writer.py [hunk 2]
```diff
@@ -186,24 +186,23 @@ def write_object_value(
...
+            self._serialize_value(temp_writer, value)
 
-            if additional_values_to_merge:
-                for additional_value in filter(lambda x: x is not None, additional_values_to_merge):
-                    self._serialize_value(temp_writer, additional_value)
-                    if on_after := self.on_after_object_serialization:
-                        on_after(additional_value)
+        for additional_value in filter(lambda x: x is not None, additional_values_to_merge):
+            self._serialize_value(temp_writer, additional_value)
+            if on_after := self.on_after_object_serialization:
```
Comment (baywet) [inline kiota_serialization_form/form_serialization_writer.py:200]: we should check for depth like dotnet does to make sure it only serializes first level objects.

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for starting this, a couple of comments.

**Review by baywet — COMMENTED** [review]
Thank you for making the changes!
I think we'll need a version bump in the toml and a changelog entry and we should be good to go here.

**Comment by shemogumbe** [comment]
> Thank you for making the changes! I think we'll need a version bump in the toml and a changelog entry and we should be good to go here.

Looks like version is dynamic, so no need to change it in toml, done so in the version.py file

**Review by baywet — APPROVED** [review]

File: kiota_serialization_form/form_serialization_writer.py [hunk 3]
```diff
@@ -186,24 +187,26 @@ def write_object_value(
...
+
+        for additional_value in filter(lambda x: x is not None, additional_values_to_merge):
+            self._serialize_value(temp_writer, additional_value)
+            if on_after := self.on_after_object_serialization:
+                on_after(additional_value)
+
+        if value and self._on_after_object_serialization:
+            self._on_after_object_serialization(value)
+
+        if len(self.writer) > 0:
```
Comment (Ndiritu) [inline kiota_serialization_form/form_serialization_writer.py:206]: Could we write unit tests for this? The `temp_writer` initialised doesn't seem to update `self.writer`. Perhaps `serialize_value` could directly write to `self`?

**Pull request merged**